feat: add support for applying existing customization specs (#2053)
hashicorp/terraform-provider-vsphere: website/docs/d/guest_os_customization.html.markdown

---
subcategory: "Virtual Machine"
layout: "vsphere"
page_title: "VMware vSphere: vsphere_guest_os_customization"
sidebar_current: "docs-vsphere-data-guest-os-customization"
description: |-

>   Provides a VMware vSphere guest customization spec data source. This can be used to apply the customization spec when virtual machine is cloned

Reviewer: ```suggestion
  Provides a VMware vSphere customization specification data source. This can be used to apply a customization specification to the guest operating system of a virtual machine after cloning.
```

---

# vsphere\_guest\_os\_customization

> The `vsphere_guest_os_customization` data source can be used to discover the details about a guest os customization

Reviewer: ```suggestion
The `vsphere_guest_os_customization` data source can be used to discover the details about a customization specification.
```

> ~> **NOTE:** The name attribute is unique identifier for the guest OS spec per VC.

Reviewer: ```suggestion
~> **NOTE:** The name attribute is the unique identifier for the customization specification per vCenter Server instance.
```

## Example Usage

```hcl
  data "vsphere_guest_os_customization" "gosc1" {

>     name          = "lin-1"

Reviewer: ```suggestion
    name          = "linux-spec"
```

  }
```

## Argument Reference

The following arguments are supported:

> * `name` - (Required) The name of the guest os customization. It is unique identifier for the guest OS customization per VC

Reviewer: ```suggestion
* `name` - (Required) The name of the customization specification is the unique identifier per vCenter Server instance.
```

> * `name` - (Required) The name of the guest os customization. It is unique identifier for the guest OS customization per VC

Reviewer: ```suggestion
* `name` - (Required) The name of the customization specification is the unique identifier per vCenter Server instance.
```

## Attribute Reference

> * `type` - The type of the guest OS. Currently Linux and Windows are supported.

Reviewer: ```suggestion
* `type` - The type of customization specification: One among: Windows, Linux.
```

> * `type` - The type of the guest OS. Currently Linux and Windows are supported.

Reviewer: ```suggestion
* `type` - The type of customization specification: One among: Windows, Linux.
```

> * `description` - The description of the Guest OS Customization spec.

Reviewer: ```suggestion
* `description` - The description for the customization specification..
```

> * `description` - The description of the Guest OS Customization spec.

Reviewer: ```suggestion
* `description` - The description for the customization specification.
```

> * `last_update_time` - The time of last modification of the Guest OS Customization spec.

Reviewer: ```suggestion
* `last_update_time` - The time of last modification to the customization specification..
```

> * `last_update_time` - The time of last modification of the Guest OS Customization spec.

Reviewer: ```suggestion
* `last_update_time` - The time of last modification to the customization specification.
```

> * `change_version` - The number of last change version of the Guest OS Customization spec.

Reviewer: ```suggestion
* `change_version` - The number of last changed version to the customization specification.
```

> * `change_version` - The number of last change version of the Guest OS Customization spec.

Reviewer: ```suggestion
* `change_version` - The number of last changed version to the customization specification.
```

> * `spec` - Container object for the Guest OS properties about to be customized . See [virtual machine customizations](#virtual-machine-customizations)

Reviewer: ```suggestion
* `spec` - Container object for the guest operating system properties to be customized. See [virtual machine customizations](#virtual-machine-customizations)
```

> * `spec` - Container object for the Guest OS properties about to be customized . See [virtual machine customizations](#virtual-machine-customizations)

Reviewer: ```suggestion
* `spec` - Container object for the guest operating system properties to be customized. See [virtual machine customizations](#virtual-machine-customizations)
```
